### ISSUE TITLE
Retweet

### ISSUE THREAD

**Issue by GoogleCodeExporter** [KGIII/pino-twitter#21]
```
As a user I would like to retweet tweets.
```

Original issue reported on code.google.com by `alex.bepple` on 21 Dec 2009 at 11:09

**Comment by GoogleCodeExporter** [comment]
```
I need ideas, how to realize this feature. Twitter API doesn't support the 
marking of 
your own retweets. So, when I retweeted some status, I can't to know, is this 
retweet 
in timeline or just simple status.
```

Original comment by `tro...@gmail.com` on 22 Dec 2009 at 11:51
- Changed state: **Accepted**
- Added labels: **Priority-Low**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
I guess it's a simple statusupdate, very similar to a reply but with the whole 
tweet.
I'm going to look at the code in the next days, to see how Vala works.
Maybe I can help a bit.
```

Original comment by `xao...@gmail.com` on 22 Dec 2009 at 9:18

**Comment by GoogleCodeExporter** [comment]
```
I mean official retweets in one click. Retweets with "RT" and so on - is just a 
simple 
status.
```

Original comment by `tro...@gmail.com` on 22 Dec 2009 at 9:22

**Comment by GoogleCodeExporter** [comment]
```
BTW I consider current situation (i.e. not having a special UI for retweet) as a
feature. Twitter has sprout a whole minilanguage in messages and it's beautiful
because you don't actually need a UI support to use them. Want a retweet: start 
your
message with "RT:". Want to quote something from real life, start with "OH:".
```

Original comment by `isagal...@gmail.com` on 27 Dec 2009 at 1:28

**Comment by GoogleCodeExporter** [comment]
```
The actual retweets I've found most do not even use. The "RT -" status is 
preferred
by most in my timeline and I. Its kind of a pain in Pino to copy the tweet, as 
right
clicking after highlighting does not bring up copy, gotta go ctrl+c, then click
reply, paste, and RT at the beginning instead of having a button. RT seems the
easiest, but I can see how the official RT function could make Pino more 
polished and
makes more sense.
```

Original comment by `FoofyFoo...@gmail.com` on 30 Dec 2009 at 3:06

**Comment by GoogleCodeExporter** [comment]
```
@troorl: I do not see why “marking” your own retweets is important. 
Primarily I was 
aiming for a simpler way to retweet than copy-paste and write “RT @…” in 
front of it.

I think the way TweetDeck does retweeting is nice: You can use both native 
retweets 
and the old-school “RT” style. You can even tell it to always use one or 
the other or 
to ask you every time you hit “Retweet”.

Splitting this into user stories gives:
1) Offer native retweeting.
2) Offer RT retweeting.
3) Offer preference for choosing one or the other.

For prioritization I would suggest: do whatever is easiest first. Later on, add 
a 
possibility to do the other thing. Preference last.

On the other hand … Preferences are quite the opposite of “simple and 
easy”. So you 
could opt for one or the other retweeting style. Personally I prefer the 
old-school 
RT style, as it also allows you to add comments to the retweet instead of 
simply 
repeating what somebody else wrote.

```

Original comment by `alex.bepple` on 30 Dec 2009 at 4:23

**Comment by GoogleCodeExporter** [comment]
```
On further thoughts I want to retract my previous comment. Retweeting in Pino 
right
now is really painful... Copying a username somehow, typing "RT:", copying a 
post
content. So I'm also +1 for a one-click feature.
```

Original comment by `isagal...@gmail.com` on 5 Jan 2010 at 2:42

**Comment by GoogleCodeExporter** [comment]
```
Ok, how about that: your click --> native retweet --> no marking in your 
timeline --> 
separate list of your own retweets with a new retweet?
```

Original comment by `tro...@gmail.com` on 5 Jan 2010 at 3:13

**Comment by GoogleCodeExporter** [comment]
```
There is no reason to fret over using the 'native' retweet functionality in the 
near
future, just implment a RT button that updates using the old-school "RT" or 
even for
some people "(via @)"
```

Original comment by `benoit.b...@gmail.com` on 7 Jan 2010 at 11:04

**Comment by GoogleCodeExporter** [comment]
```
Also, the priority of this issue should higher considering it's a central piece 
of
Twitter functionality.
```

Original comment by `benoit.b...@gmail.com` on 7 Jan 2010 at 11:05

**Comment by GoogleCodeExporter** [comment]
Original comment by `tro...@gmail.com` on 11 Jan 2010 at 6:11
- Added labels: **Priority-High**
- Removed labels: **Priority-Low**

**Comment by GoogleCodeExporter** [comment]
```
Closed by b4cc79cd0a revision.
```

Original comment by `tro...@gmail.com` on 12 Jan 2010 at 3:11
- Changed state: **Fixed**
